src/components/activity/useActivityNotifiction.ts

```diff
@@ -0,0 +1,27 @@
+import { useWeb3 } from 'libs/web3';
+import { useActivityQuery } from './useActivityQuery';
+import { useEffect, useState } from 'react';
+import { useNotifications } from 'hooks/useNotifications';
+
+export const useActivityNotifications = () => {
+  const { user } = useWeb3();
+  const [previous, setPrevious] = useState<number | null>(null);
+  const query = useActivityQuery({ ownerId: user });
+  const allActivities = query.data || [];
+  const buyOrSell = allActivities.filter(
+    (a) => a.action === 'sell' || a.action === 'buy'
+  );
+  const { dispatchNotification } = useNotifications();
+
+  useEffect(() => {
+    if (query.isLoading) return;
```

Comment:
this looks like something that the query callbacks would handle great - wanna jump on a quick session to explore if we can refactor this to remove the useEffect and use the useQuery callbacks?